### ISSUE TITLE
MediaEncoder + ffmpeg behaviour

### ISSUE THREAD

**Issue by DomHoney** [MediaBrowser/Emby#3269]
Hi,

First of all thanks to the Emby team for all their hard work on this project, its really appreciated.

I am undertaking a personal project where I am interested in the behaviour of `Emby.Server.MediaEncoding`, specifically how it interacts with ffmpeg to detect when the (trans|en)coding output has been processed and is ready for consumption.

I understand that the code for this component is closed source, I am just wondering if anyone can provide some specific details of a small aspect of its behaviour.

For context, my project's purpose is to offload transcoding to a 'worker' machine by intercepting calls to the master ffmpeg binary, running ffmpeg on the worker and then sending the output (stderr + files) back to the master where Emby is running. I understand this is ambitious and presents a minefield of problems, but I have implemented a decent amount so far.

What I have done up to now is to try and reverse engineer the DLL by monitoring the transcoding directory with `inotifywait -m <transcoding-temp>` to understand how ffmpeg produces its output given a particular set of input/output parameters.

I have successfully been able to reproduce the filesystem events on the master using Python; opening, modifying, and closing segment files as ffmpeg does when configured to output in segment format - Emby does not seem to pick up the files for playback.

With my limited knowledge, I can only guess that Emby looks for these new files when ffmpeg outputs the following via stderr for each file:

```[segment @ 0x562307a45300] Opening '/var/lib/emby-server/transcoding-temp/e215d18f9be7e644b34cf40d2a0833f9.m3u8.tmp' for writing\n```

Unfortunately the code I've written will need to be uprooted in order to implement logic that synchronizes this stderr output with the segment file writing, so I want to be sure before attempting to do so.

I realise I've written a wall of text and likely over-explained myself, but I was hoping someone could let me know if I'm on the right track. A brief explanation of how exactly `Emby.Server.MediaEncoding` picks up ffmpeg's output would be a great help.

Thanks,

Dom

**Comment by softworkz** [comment]
This doesn't seem to be a question related to using Emby.

It's rather a question about how to use ffmpeg and the right place for such questions is https://ffmpeg.org/

**Comment by DomHoney** [comment]
The question isn't about how ffmpeg works - I know how it works because it is open source, not to mention I can monitor its activity with `inotifywait` as I've already stated.

The question is asking how the closed source binary `Emby.Server.MediaEncoding` processes ffmpeg's output.

But from what I gathered in other issues, the proprietary blobs are to be removed in the future anyway, so this can be left closed.
